src/test/java/org/mockito/internal/configuration/InjectingAnnotationEngineTest.java

```diff
@@ -1,55 +0,0 @@
-/*
- * Copyright (c) 2019 Mockito contributors
- * This program is made available under the terms of the MIT License.
- */
-package org.mockito.internal.configuration;
-
-import org.junit.Assert;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.Spy;
-import org.mockito.junit.MockitoJUnitRunner;
-
-class I {}
-
-@RunWith(MockitoJUnitRunner.class)
-public class InjectingAnnotationEngineTest extends I {
```

Comment:
As a user, I think the behavior as described in this test makes sense. Therefore, I would like to keep it. That said, it would be ideal if we can keep this behavior as-is, while also fixing your issue. Do you think that is feasible or are they mutually exclusive?

Reply:
1. This test tests a behaviour that was never there. It tests a bug. The proper behaviour for this would need to be implemented.
2. While this behaviour might make sense in some cases, it might be highly problematic in other cases. So I strongly advise against this feature. I could very well imagine this breaking quite some use cases of other's. The only reason nobody complained yet about this "feature"/tested behaviour that is does not exist yet(only in edge cases and even there does the wrong thing). Proper way of doing this would probably be to annotate setters used for DI explicitly. But I think that is a little to much to code into Mockito. 
3. The docs explicitly state the opposite of what this test is doing.

For more details/examples please read the comments in the issue.

Reply:
Maybe there is a clever way to combine all use cases here but it would need to be implemented. There is nothing there yet as one might assume by this test

Reply:
Commonly Beans contain fields that are not supposed to be injected, maybe an object field even. How would one know if a field is supposed to be injected? Well only way to let the user make annotations right?

A lot of strange cases can arise here, I would not go down that road. That's why DI Frameworks have annotations to clarify all this. And commonly constructor injection is recommended.

Reply:
Injection in Mockito has always been a difficult topic. What one considers a bug, others consider a feature. In this case, the current behavior is what some of our users are relying on, as you saw with the reverts that happened. Therefore, if we were to make changes again, we are breaking those users.

This means that the Mockito maintainers now have to choose between two scenarios, without being able to satisfy all users. In this case, I lean towards keeping the current behavior to be conservative, rather than breaking these users and introduce regressions in their test suites.

If you want fully deterministic dependency injection, I advise you to use a DI framework as you mentioned. Mockito injection is a best-effort attempt.

Reply:
Yes but current state is just a mess

Reply:
Whatever you do leaving it like it is know is super broken

Reply:
If I were you I would not support cases of mixed injection, where both constructor and setter injection is being used

Reply:
I am still not sure whether this behavior makes sense or not. @tokuhirom since you contributed the original regression test, can you please review this PR?